Lib/Inc/ICommand.h

```diff
@@ -21,14 +21,14 @@
 class ICommand
 {
 public:
-    ICommand();
+    ICommand() {}
 
     /**
      * @brief Get name of the last parsed command.
      *
      * @return Name of the last parsed command.
      */
-    virtual etl::string<MAX_STRING_SIZE> *getName() const = 0;
+    virtual etl::string<MAX_STRING_SIZE> getName() const = 0;
```

Comment:
This will break a lot of code. Why do you want to return by value?

Reply:
The way this method is right now doesn't work if I store the name in a `etl::string<MAX_STRING_SIZE> m_command_name;` variable. 
The ways I could make it work were by either removing the pointer or by changing it to: 
`virtual const etl::string<MAX_STRING_SIZE> getName() const = 0` . Perhaps this is a better option.